app_build_suite/build_steps/helm.py

```diff
@@ -382,6 +403,72 @@ def pre_run(self, config: argparse.Namespace) -> None:
                 ):
                     raise ValidationError(self.name, f"Value of '{option}' is not a correct boolean.")
 
+    def additional_annotations(self, chart_file_name: str, catalog_name: str) -> Dict[str, Any]:
+        catalog_base_url = f"{self._value_annotation_catalog_base_url}{catalog_name}/{chart_file_name}-meta/"
+        annotations = {}
+        # TODO check if these exist
+        annotations[self._key_annotation_metadata_url] = f"{catalog_base_url}main.yaml"
+        annotations[self._key_annotation_readme_url] = f"{catalog_base_url}README.md"
+        return annotations
+
+    def run(self, config: argparse.Namespace, context: Dict[str, Any]) -> None:
+        if not config.generate_metadata:
+            logger.info("Metadata generation is disabled using 'generate-metadata' option.")
+            return
+        # read current Chart.yaml
+        chart_yaml_path = os.path.join(config.chart_dir, _chart_yaml)
+        with open(chart_yaml_path, "r") as file:
+            chart_yaml = yaml.safe_load(file)
+        # try to guess the package file name. we need it for url generation in annotations
+        chart_name = os.path.basename(config.chart_dir)
+        context[context_key_chart_file_name] = f"{chart_name}-{context[context_key_git_version]}.tgz"
+        # put in generated annotations
+        chart_yaml[self._key_annotations] = {
```

Comment:
I'm not sure what you mean here... are you trying to insert an empty `annotations` dict if it doesn't exist? If yes, I think something like this is more readable
```python
if self._key_annotations not in chart_yaml.keys:
  chart_yaml[_key_annotations] = {}
```

Reply:
This makes sure settings `annotations` in the `chart_yaml` succeeds even when the original `chart_yaml` has no `annotations` while also overwriting annotations generated in `additional_annotations()`.